src/commands/highlights.rs

```diff
@@ -0,0 +1,96 @@
+use super::*;
+use crate::Arc;
+
+#[command("highlights")]
+#[sub_commands(highlights_add, highlights_get, highlights_remove)]
+#[usage("!highlights <add | get | remove>")]
+pub async fn highlights(_: &client::Context, _: &Message) -> CommandResult {
+    abort_with!(UserErr::invalid_usage(&HIGHLIGHTS_COMMAND_OPTIONS))
+}
+
+/// add a highlight for your user
+#[command("add")]
+#[usage("!highlights add <word>")]
+pub async fn highlights_add(ctx: &client::Context, msg: &Message, args: Args) -> CommandResult {
+    let args = args.message().trim().to_string();
+    if args.contains(" ") {
+        abort_with!(UserErr::Other("Highlight can't contain a space for implementation/performance reasons".to_string()));
+    } else if args.is_empty() {
+        abort_with!(UserErr::InvalidUsage("You must provide a argument"));
+    } else if args.len() < 3 {
+        abort_with!(UserErr::Other("highlight has to be larger than 2 characters".to_string()));
+    }
+
+    let db: Arc<Db> = ctx.get_db().await;
+    let n: u8 = match crate::checks::get_permission_level(ctx, msg)
+        .await
+        .unwrap_or(PermissionLevel::User)
+    {
+        PermissionLevel::Mod => 20,
+        _ => 4,
+    };
+
+    let highlights = db.get_highlights().await?;
+    let highlights_by_user = highlights.iter()
+        .filter(|(_, users)| users.contains(&msg.author.id))
+        .map(|(word, _)| word);
+
+    if highlights_by_user.collect_vec().len() as u8 == n {
+        abort_with!(UserErr::Other(format!("You already set your {} highlights", n)));
+    }
+    db.set_highlight(msg.author.id, args.clone())
+        .await
+        .user_error("Something went wrong")?;
```

Comment:
given that you're only fetching the highlights from the db to check the max-count here, this is a lot of unnecessary overhead.
A better implementation would be to have a specific query for the count of highlights of a user, and check that
```sql
SELECT COUNT(*) FROM highlights WHERE usr=?
```

Reply:
given that the highlights are being cached, it _may_ be okay to not do it in a Db query ever - as the db query would not really be used much. in that case still the count should be a method in `Db` tho

Reply:
That was my thought process. But I felt like it just added complexity to have a seperate function, when it's really not hard to count something.

Reply:
I really don't think that counting this will be used anywhere but here... Splitting this up won't help with anything. 

Mind if I just leave it like this?

Reply:
I see you agree. Epic